muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/communicator/CommunicatorAutoReply.java

```diff
@@ -0,0 +1,112 @@
+
+
+    if ((communicatorMessage != null) && (sender != null) && (recipient != null)) {
+      if (!Objects.equals(sender.getId(), recipient.getId())) {
+
+        if (autoReply != null) {
+          
+          if (autoReply.getValue().equals("ENABLED")) {
+            UserEntityProperty startProperty = userEntityController.getUserEntityPropertyByKey(recipient, "profile-vacation-start");
+            UserEntityProperty endProperty = userEntityController.getUserEntityPropertyByKey(recipient, "profile-vacation-end");
+  
+            LocalDate start = null;
+            LocalDate end = null;
+            String replyMessage = "";    
+            String replySubject = "";
+            
+            if (startProperty != null && endProperty != null) {
+              DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'", Locale.ENGLISH);
+              start = LocalDate.parse(startProperty.getValue(), formatter);
+              end = LocalDate.parse(endProperty.getValue(), formatter);
+            }
+            if (start != null && end != null) {  
+              LocalDate now = LocalDate.now();
+              Date notificationDate = new Date();
+              if (start.equals(now) || (now.isAfter(start) && now.isBefore(end)) || end.equals(now)) {
+                
+                DateTimeFormatter outputFormatter = DateTimeFormatter.ofPattern("dd.MM.yyy", Locale.ENGLISH);
+                String startDate = outputFormatter.format(start);
+                String endDate = outputFormatter.format(end);
+                
+                if (autoReplyMsg != null) {
+                  replyMessage = autoReplyMsg.getValue();
+                } else {
+                  replyMessage = "Automaattinen vastaus: Poissa " + startDate + " - " + endDate;
+                }
+                if (autoReplySubject != null) {
+                  replySubject = autoReplySubject.getValue();
+                } else {
+                  replySubject = "Automaattinen vastaus: Poissa " + startDate + " - " + endDate;
+                }
+                List<UserEntity> recipientsList = new ArrayList<UserEntity>();
+                recipientsList.add(sender);
+                VacationNotifications vacationNotification = communicatorController.findVacationNotification(sender, recipient);
+                
+                if (vacationNotification != null) {
+                  Date lastNotification = vacationNotification.getNotificationDate();
+
+                  LocalDate lastPlusFour = lastNotification.toInstant().atZone(ZoneId.systemDefault()).toLocalDate();
+
+                  lastPlusFour = lastPlusFour.plusDays(4);
+                  Date fourDays = java.sql.Date.valueOf(lastPlusFour);
+
+                  if (java.sql.Date.valueOf(now).after(fourDays)) {
+                    communicatorController.createVacationNotification(sender, recipient, notificationDate);
```

Comment:
This needs to be an update method for the existing VacationNotification entity. The table has been structured so that each sender+recipient combo should only appear once so if you add a new vacation notification for the same pair, the DAO finder method will crash the next time it is called.